<?xml version="1.0" encoding="UTF-8"?>
<interface>
  <!-- interface-requires gtk+ 3.24 -->
  <template class="IdeCloneSurface" parent="IdeSurface">
    <child>
      <object class="GtkScrolledWindow">
        <property name="propagate-natural-width">true</property>
        <property name="propagate-natural-height">true</property>
        <property name="hscrollbar-policy">never</property>
        <property name="visible">true</property>
        <child>
          <object class="GtkViewport">
            <property name="visible">true</property>
            <child>
              <object class="GtkBox">
                <property name="margin">32</property>
                <property name="orientation">vertical</property>
                <property name="valign">start</property>
                <property name="vexpand">true</property>
                <property name="visible">true</property>
                <child>
                  <object class="DzlThreeGrid" id="grid">
                    <property name="column-spacing">12</property>
                    <!-- can't use row-spacing because we have to animate in
                         the revealer which messes up the margins.  -->
                    <property name="row-spacing">0</property>
                    <property name="expand">true</property>
                    <property name="visible">true</property>
                    <child>
                      <object class="GtkImage" id="splash">
                        <property name="valign">end</property>
                        <property name="vexpand">true</property>
                        <property name="icon-name">document-save-symbolic</property>
                        <property name="pixel-size">128</property>
                        <property name="visible">true</property>
                        <property name="margin">24</property>
                        <style>
                          <class name="dim-label"/>
                        </style>
                      </object>
                      <packing>
                        <property name="row">0</property>
                        <property name="column">1</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkLabel" id="repo_label">
                        <property name="label" translatable="yes">Repository URL</property>
                        <property name="valign">center</property>
                        <property name="visible">true</property>
                        <property name="xalign">1.0</property>
                        <style>
                          <class name="dim-label"/>
                        </style>
                      </object>
                      <packing>
                        <property name="row">1</property>
                        <property name="column">0</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkLabel" id="uri_entry_help">
                        <property name="label" translatable="yes">Enter the repository of the project you would like to clone. The URL should look similar to "https://gitlab.gnome.org/GNOME/gnome-builder.git".</property>
                        <property name="margin-top">3</property>
                        <property name="width-chars">40</property>
                        <property name="max-width-chars">60</property>
                        <property name="visible">true</property>
                        <property name="wrap">true</property>
                        <property name="xalign">0.0</property>
                        <style>
                          <class name="dim-label"/>
                        </style>
                        <attributes>
                          <attribute name="scale" value="0.833333"/>
                        </attributes>
                      </object>
                      <packing>
                        <property name="row">2</property>
                        <property name="column">1</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkEntry" id="uri_entry">
                        <property name="placeholder-text" translatable="yes">user@host:repository.git</property>
                        <property name="width-chars">40</property>
                        <property name="max-width-chars">50</property>
                        <property name="valign">center</property>
                        <property name="visible">true</property>
                        <signal name="changed" handler="ide_clone_surface_uri_entry_changed" swapped="true" object="IdeCloneSurface"/>
                      </object>
                      <packing>
                        <property name="row">1</property>
                        <property name="column">1</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkToggleButton" id="more_button">
                        <property name="halign">start</property>
                        <property name="hexpand">false</property>
                        <property name="visible">true</property>
                        <property name="has-tooltip">true</property>
                        <property name="tooltip-text" translatable="yes">Select branch and other options.</property>
                        <style>
                          <class name="flat"/>
                        </style>
                        <child>
                          <object class="GtkImage">
                            <property name="visible">true</property>
                            <property name="icon-name">view-more-symbolic</property>
                            <style>
                              <class name="image-button"/>
                            </style>
                          </object>
                        </child>
                      </object>
                      <packing>
                        <property name="row">1</property>
                        <property name="column">2</property>
                      </packing>
                    </child>
                  </object>
                </child>
                <child>
                  <object class="GtkRevealer" id="more_revealer">
                    <property name="reveal-child" bind-source="more_button" bind-property="active"/>
                    <property name="visible">true</property>
                    <child>
                      <object class="GtkBox">
                        <property name="margin-top">12</property>
                        <property name="orientation">vertical</property>
                        <property name="visible">true</property>
                        <child>
                          <object class="DzlThreeGrid">
                            <property name="column-spacing">12</property>
                            <property name="row-spacing">12</property>
                            <property name="visible">true</property>
                            <child>
                              <object class="GtkLabel" id="kind_label">
                                <property name="label" translatable="yes">Repository Kind</property>
                                <property name="visible">false</property>
                                <property name="xalign">1.0</property>
                                <property name="valign">center</property>
                                <style>
                                  <class name="dim-label"/>
                                </style>
                              </object>
                              <packing>
                                <property name="row">0</property>
                                <property name="column">0</property>
                              </packing>
                            </child>
                            <child>
                              <object class="DzlRadioBox" id="kind_radio">
                                <property name="visible">false</property>
                                <property name="valign">center</property>
                              </object>
                              <packing>
                                <property name="row">0</property>
                                <property name="column">1</property>
                              </packing>
                            </child>
                            <child>
                              <object class="GtkLabel" id="branch_label">
                                <property name="label" translatable="yes">Branch</property>
                                <property name="visible">true</property>
                                <property name="xalign">1.0</property>
                                <property name="valign">center</property>
                                <style>
                                  <class name="dim-label"/>
                                </style>
                              </object>
                              <packing>
                                <property name="row">1</property>
                                <property name="column">0</property>
                              </packing>
                            </child>
                            <child>
                              <object class="GtkEntry" id="branch_entry">
                                <property name="visible">true</property>
                                <property name="valign">center</property>
                                <property name="text">master</property>
                              </object>
                              <packing>
                                <property name="row">1</property>
                                <property name="column">1</property>
                              </packing>
                            </child>
                            <child>
                              <object class="GtkLabel" id="author_label">
                                <property name="label" translatable="yes">Author Name</property>
                                <property name="visible">true</property>
                                <property name="xalign">1.0</property>
                                <property name="valign">center</property>
                                <style>
                                  <class name="dim-label"/>
                                </style>
                              </object>
                              <packing>
                                <property name="row">2</property>
                                <property name="column">0</property>
                              </packing>
                            </child>
                            <child>
                              <object class="GtkEntry" id="author_entry">
                                <property name="visible">true</property>
                                <property name="valign">center</property>
                              </object>
                              <packing>
                                <property name="row">2</property>
                                <property name="column">1</property>
                              </packing>
                            </child>
                            <child>
                              <object class="GtkLabel" id="email_label">
                                <property name="label" translatable="yes">Author Email</property>
                                <property name="visible">true</property>
                                <property name="xalign">1.0</property>
                                <property name="valign">center</property>
                                <style>
                                  <class name="dim-label"/>
                                </style>
                              </object>
                              <packing>
                                <property name="row">3</property>
                                <property name="column">0</property>
                              </packing>
                            </child>
                            <child>
                              <object class="GtkEntry" id="email_entry">
                                <property name="visible">true</property>
                                <property name="valign">center</property>
                              </object>
                              <packing>
                                <property name="row">3</property>
                                <property name="column">1</property>
                              </packing>
                            </child>
                          </object>
                        </child>
                      </object>
                    </child>
                  </object>
                </child>
                <child>
                  <object class="DzlThreeGrid">
                    <property name="margin-top">12</property>
                    <property name="column-spacing">12</property>
                    <property name="visible">true</property>
                    <child>
                      <object class="GtkLabel" id="dest_label">
                        <property name="label" translatable="yes">Project Destination</property>
                        <property name="visible">true</property>
                        <property name="xalign">1.0</property>
                        <property name="valign">center</property>
                        <style>
                          <class name="dim-label"/>
                        </style>
                      </object>
                      <packing>
                        <property name="row">0</property>
                        <property name="column">0</property>
                      </packing>
                    </child>
                    <child>
                      <object class="DzlFileChooserEntry" id="destination_chooser">
                        <property name="valign">center</property>
                        <property name="visible">true</property>
                        <signal name="notify::file" handler="ide_clone_surface_destination_changed" object="IdeCloneSurface" swapped="true"/>
                      </object>
                      <packing>
                        <property name="row">0</property>
                        <property name="column">1</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkLabel" id="destination_label">
                        <property name="visible">true</property>
                        <property name="xalign">0.0</property>
                        <property name="valign">center</property>
                        <property name="margin-top">3</property>
                        <style>
                          <class name="dim-label"/>
                        </style>
                        <attributes>
                          <attribute name="scale" value="0.83333"/>
                        </attributes>
                      </object>
                      <packing>
                        <property name="row">2</property>
                        <property name="column">1</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkBox">
                        <property name="margin-top">32</property>
                        <property name="visible">true</property>
                        <property name="orientation">horizontal</property>
                        <child>
                          <object class="GtkStack" id="button_stack">
                            <property name="visible">true</property>
                            <property name="homogeneous">true</property>
                            <property name="halign">end</property>
                            <child>
                              <object class="GtkButton" id="clone_button">
                                <property name="label" translatable="yes">Clone Project</property>
                                <property name="visible">true</property>
                                <signal name="clicked" handler="ide_clone_surface_clone" object="IdeCloneSurface" swapped="true"/>
                                <style>
                                  <class name="suggested-action"/>
                                </style>
                              </object>
                            </child>
                            <child>
                              <object class="GtkButton" id="cancel_button">
                                <property name="label" translatable="yes">Cancel</property>
                                <property name="visible">true</property>
                                <style>
                                  <class name="destructive-action"/>
                                </style>
                              </object>
                            </child>
                          </object>
                          <packing>
                            <property name="pack-type">end</property>
                          </packing>
                        </child>
                      </object>
                      <packing>
                        <property name="row">3</property>
                        <property name="column">1</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkLabel" id="status_message">
                        <property name="margin-top">24</property>
                        <property name="valign">center</property>
                        <property name="visible">true</property>
                        <property name="width-chars">50</property>
                        <property name="max-width-chars">50</property>
                        <property name="wrap">true</property>
                        <property name="xalign">0.0</property>
                        <style>
                          <class name="dim-label"/>
                        </style>
                      </object>
                      <packing>
                        <property name="row">4</property>
                        <property name="column">1</property>
                      </packing>
                    </child>
                  </object>
                </child>
              </object>
            </child>
          </object>
        </child>
      </object>
    </child>
  </template>
  <object class="GtkSizeGroup">
    <property name="mode">horizontal</property>
    <widgets>
      <widget name="splash"/>
      <widget name="branch_entry"/>
      <widget name="author_entry"/>
      <widget name="email_entry"/>
      <widget name="uri_entry"/>
      <widget name="uri_entry_help"/>
      <widget name="destination_chooser"/>
      <widget name="kind_radio"/>
    </widgets>
  </object>
  <object class="GtkSizeGroup">
    <property name="mode">horizontal</property>
    <widgets>
      <widget name="repo_label"/>
      <widget name="dest_label"/>
      <widget name="author_label"/>
      <widget name="email_label"/>
      <widget name="branch_label"/>
    </widgets>
  </object>
</interface>
